### PR TITLE
[Cu-jd87v6] Update creator to schedule matches

### PR DESCRIPTION
**Pull request by eastemployeer**
Checked, working. 
Note copied from our FB conversation:

>  Te daty ktore leca z frontu sa w formacie Date czyli czasu lokalnego czyli srodkowo europejskiego (CET). No a daty w bazie przechowywane sa jako ISO string. I to jest format ktory jest cofnięty o 2h do tyle. Czyli jak z frontu idzie np. 08.06 00:00:00 to w bazie (ISO string) zapisuje sie to jako 07.06 22:00:00

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-jd87v6 Update creator to schedule matches](https://app.clickup.com/t/jd87v6)

File: server/concepts/Tournament.ts [hunk 1]
```diff
@@ -202,23 +213,78 @@ export default class Tournament implements TournamentWithoutMS {
...
+    // but if there are none, we'll return the first valid day of next week (again, assuming the days are sorted)
+    return thisWeek || moment(currentDate).add(1, 'weeks').isoWeekday(daysArray[0]);
+  }
+
+  private static generateRoundRobinMatches(teams: Required<TeamWithoutMS>[],
+    matchesDateInfo: MatchesDateInfo): Match[] {
     const { data } = tournamentGenerator([...teams], { type: 'single-round' });
 
+    const days = matchesDateInfo.matchDays.sort((a: number, b: number) => a - b);
+    let currentDate = matchesDateInfo.startingDate;
```
Comment (Zummek) [inline server/concepts/Tournament.ts:298]: I'm not sure if we should assign startingDate to `currentDate`. that can be a bit confusing. I think better use `matchesDateInfo.startingDate` object or use destructuring assignment for `startingDate`

File: server/concepts/Tournament.ts [hunk 2]
```diff
@@ -202,23 +213,78 @@ export default class Tournament implements TournamentWithoutMS {
     await TournamentRepository.delete(tournamentId);
   }
 
-  private static generateRoundRobinMatches(teams: Required<TeamWithoutMS>[]): Match[] {
+  public static isThisDateInFuture(targetDayNum: number, currentDate: Date) {
```
Comment (Zummek) [inline server/concepts/Tournament.ts:262]: The name of this function is not appropriate for her operation. The name of that function should be like "will this day of week be in this week".
 ofc if I understand well how it works

File: server/concepts/Tournament.ts [hunk 3]
```diff
@@ -202,23 +213,78 @@ export default class Tournament implements TournamentWithoutMS {
...
+      const nextMatchDate = new Date(nextMatchDateMoment.toISOString());
+
+      if (frequencyCounter >= matchesDateInfo.frequency) {
+        frequencyCounter = 0;
+        currentDate = nextMatchDate;
+      }
+      return Match.getNewInstance({
+        teamA: match.homeTeam,
+        teamB: match.awayTeam,
+        date: nextMatchDate,
```
Comment (Zummek) [inline server/concepts/Tournament.ts]:
```suggestion
      const nextMatchDateMoment = Tournament.findNextInstanceInDaysArray(days, currentDate);
      const nextMatchDate = nextMatchDateMoment.toDate();

      if (frequencyCounter >= matchesDateInfo.frequency) {
        frequencyCounter = 0;
        currentDate = nextMatchDate;
      }
      return Match.getNewInstance({
        teamA: match.homeTeam,
        teamB: match.awayTeam,
        date: nextMatchDate,
```

File: server/concepts/Tournament.ts (hunk 2 above)
Comment (Zummek) [inline server/concepts/Tournament.ts:262]: I think we can move this function to helpers functions/files

File: server/concepts/Tournament.ts [hunk 4]
```diff
@@ -202,23 +213,78 @@ export default class Tournament implements TournamentWithoutMS {
...
 
     for (let i = 0; i < matchAmount; i++) {
       newMatches.push(Match.getNewInstance());
     }
-    // generowanie dat
+    for (let i = matchAmount - 1; i >= 0; i--) {
+      frequencyCounter++;
+
+      const nextMatchDateMoment = Tournament.findNextInstanceInDaysArray(days, currentDate);
+      const nextMatchDate = new Date(nextMatchDateMoment.toISOString());
```
Comment (Zummek) [inline server/concepts/Tournament.ts:345]:
```suggestion
      const nextMatchDate = nextMatchDateMoment.toDate();
```

File: server/concepts/Tournament.ts (hunk 2 above)
Comment (Zummek) [inline server/concepts/Tournament.ts:262]: I think we can move this function to helpers functions/files

**Review by Zummek — APPROVED** [review]
Logic of this code is okej but I have some suggestions to naming

File: server/concepts/Tournament.ts (hunk 1 above)
Comment (eastemployeer) [reply server/concepts/Tournament.ts:298]: You didn't analyze code appropriately. You don't know what is it doing.

Comment (Zummek) [reply server/concepts/Tournament.ts:298]: I think startingDate is the first day of tournament. But `currentDate` confuses me with current date like now 20:30 05.06.2021, not future date

**Pull request merged**